src/util/governance.h

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2014-2023 The Dash Core developers
+// Distributed under the MIT/X11 software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_UTIL_GOVERNANCE_H
+#define BITCOIN_UTIL_GOVERNANCE_H
+
+#include <primitives/transaction.h>
+#include <uint256.h>
+
+#include <serialize.h>
+
+#include <string>
+#include <vector>
+
+/**
+ * This module is a public interface of governance module that can be used
+ * in other components such as wallet
+ */
+
+class UniValue;
+
+enum class GovernanceObject : int {
+    UNKNOWN = 0,
+    PROPOSAL,
+    TRIGGER
+};
+template<> struct is_serializable_enum<GovernanceObject> : std::true_type {};
+
+namespace Governance
+{
+struct Object
```

Comment:
this should be a class; while basically the same; we should stick to structs simply store some values; while classes also have methods and are generally more complex

Reply:
agree, here's no complex logic but still 2 constructors and several helpers